src/libutil/logging.cc

```diff
@@ -190,7 +190,7 @@ struct JSONLogger : Logger {
 
     void write(const nlohmann::json & json)
     {
-        prevLogger.log(lvlError, "@nix " + json.dump(-1, ' ', false, nlohmann::json::error_handler_t::replace));
+        writeLine(fd, "@nix " + json.dump(-1, ' ', false, nlohmann::json::error_handler_t::replace));
```

Comment:
This is in general not atomic, but works out for writes under 1 page size on Linux.

Reply:
Right, but it wasn't atomic before either. Still, we should probably add a mutex or something like that.